app/models/term.rb

```diff
@@ -67,6 +86,36 @@ def calculate_categorizations
 
   private
 
+  # The store_fingerprint method gets called before a Term record is saved, ensuring that Terms should always have a
+  # related TermFingerprint method.
+  def store_fingerprint
+    self.term_fingerprint = TermFingerprint.find_or_create_by({ fingerprint: calculate_fingerprint })
+  end
+
+  # This is similar to the SuggestedResource fingerprint method, with the exception that it also replaces &quot; with "
+  # during its operation. This switch may also need to be added to the SuggestedResource method, at which point they can
+  # be abstracted to a helper method.
+  def calculate_fingerprint
```

Comment:
I don't see any tests that confirm this logic works as expected.

You can likely snag them from https://github.com/MITLibraries/tacos/blob/main/test/models/detector/suggested_resource_test.rb
as we'll be refactoring and removing those tests soon to use the Term fingerprints.

This feels too important to hide behind "we don't test private methods" to me. If you don't test it directly (which is fine for a private method), we still should test the effects of it by testing the generated fingerprints on a Term (although it's probably slightly easier to just test it directly).

I'm also curious if this might better belong in the Fingerprint model itself. It feels slightly odd to have a model dedicated to fingerprints but have the actual logic in Term.

Reply:
Thankfully, ending up with an instance method for this logic means that we can't hide behind "we don't test private methods" anymore. Either way, I agree that tests are needed here - not sure why I didn't write them initially.

I've tried to make clear in the test structure that there's a test for every line in the method, to make sure we keep focus on its components rather than trying to manage one all-encompassing example. Hopefully future-us will continue to keep up if this logic changes.